S3FileRenamer.py

```diff
@@ -23,18 +21,33 @@ def contains_bad_characters(string):
     else:
         return False, ""
 
-
 def perform_rename(old_name, root_path, found_bad_characters, rename_do):
     """Remove illegal characters if rename argument was passed to program."""
-    corrected_name = re.sub("|".join(found_bad_characters), "", old_name)
+# Replaced "re.sub" because of conflict with regular expression wildcards
```

Comment:
Test case?

Reply:
In particular re.sub() has trouble with the "$" character. It also had a problem with "." .

Reply:
More explicitly, that method made it easy accidentally to construct an illegal regex for `re.sub`.  The regex was constructed by joining all the "bad" filename characters with `|` in between them.  Unfortunately, several of the potential "bad" characters have themselves special meaning in regexes.  This can lead to `"|".join(found_bad_characters)` resulting in a syntactically invalid regex.  E.g., `found_bad_characters` consisting of `+` and `]` would result in the aforementioned expression yielding a regex of `+|]`, which is invalid regex syntax and would throw an exception.

Even the filename `lost+found` would yield an illegal regex because the resultant `+` contains a suffix operator applying to nothing.

Actually, the `$` and `.` "bad" characters are less catastrophic by comparison. 😃